lib/xnvme_be_macos_driverkit_admin.c

```diff
@@ -0,0 +1,61 @@
+// SPDX-FileCopyrightText: Samsung Electronics Co., Ltd
+//
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include <libxnvme.h>
+#include <xnvme_be.h>
+#include <xnvme_be_nosys.h>
+#ifdef XNVME_BE_MACOS_ENABLED
+#include <xnvme_be_macos_driverkit.h>
+#include <mach/mach_error.h>
+
+#include <xnvme_dev.h>
+
+int
+xnvme_be_macos_driverkit_admin(struct xnvme_cmd_ctx *ctx, void *dbuf, size_t dbuf_nbytes,
+			       void *XNVME_UNUSED(mbuf), size_t XNVME_UNUSED(mbuf_nbytes))
+{
+	struct xnvme_be_macos_driverkit_state *state =
+		(struct xnvme_be_macos_driverkit_state *)ctx->dev->be.state;
+
+	uint64_t _buf_base_offset = 0;
+	if (dbuf) {
+		struct buffer *buf = buffer_find(state->buffers, (uint64_t)dbuf);
+		assert(buf);
+		_buf_base_offset = ((uint64_t)dbuf) - buf->vaddr;
+	}
+	NvmeSubmitCmd cmd = {
+		.queue_id = 0,
+		.dbuf_nbytes = dbuf_nbytes,
+		.dbuf_offset = _buf_base_offset,
+		.dbuf_token = (uint64_t)dbuf - _buf_base_offset,
+	};
+	NvmeSubmitCmd cmd_return;
+	memcpy(cmd.cmd, &ctx->cmd, sizeof(struct xnvme_spec_cmd));
+	cmd.backend_opaque = (uint64_t)ctx;
+
+	size_t output_cnt = sizeof(NvmeSubmitCmd);
+	kern_return_t ret =
```

Comment:
When declaring variables, then the initialization done must be:

* Variable unpacking, such as
  - ``struct xnvme_be_macos_driverkit_state *state = (struct xnvme_be_macos_driverkit_state *)ctx->dev->be.state``
* Static initialization, such as
  - ``uint64_t _buf_base_offset = 0;``
  
Things that require error-handling, come after this. Ordering at the top-scope should be:

* Variable declaration with initialization using argument unpacking
* Variable declaration with static initializer
* Variable declaration without initialization

Reply:
Done.